packages/adapters/da/src/lib.rs

```diff
@@ -0,0 +1,28 @@
+use std::time::Duration;
+
+pub enum DALayerError {
+    Error,
+}
+
+pub enum SubmissionStatus {
+    Pending,
+    Completed,
+    Finalized,
+    Failed,
+}
+
+pub trait DALayer: Send + Sync {
+    fn post_data(
+        &self,
+        data: Vec<u8>,
+    ) -> impl std::future::Future<Output = Result<Vec<u8>, DALayerError>> + Send;
+    fn check_status(
+        &self,
+        submission_id: Vec<u8>,
```

Comment:
nit: may want to introduce a type for the submission id to ease readability/usage

```
/// Represents a submission ID returned by the DA layer
pub type SubmissionId = Vec<u8>;
```
and use in code like so:
```suggestion
        data: Vec<u8>,
    ) -> impl std::future::Future<Output = Result<SubmissionId, DALayerError>> + Send;
    fn check_status(
        &self,
        submission_id: SubmissionId,
```